### DISCUSSION TITLE
File transfer question

### DISCUSSION THREAD

**Discussion by schy3k** [FreeRDP/FreeRDP#7967] [Q&A]
HI

macos development

Is there a way to transfer files during rdp connection?

Are there any other channels besides the clipboard channel?

**Comment by akallabeth** [comment]
1. you can use drive redirection to copy files
2. yes, drive, smartcard and many others, check the source tree (or man page, ...) for documentation

**Reply by schy3k** [reply]
Drag and drop in development

Can I copy specific files from the client to a specific location on the server?

I want to automatically transfer files.

**Comment by akallabeth** [comment]
short answer: no

longer one:
1. You can use the file clipboard to allow pasting on the server side (but paste is still something you need to trigger server side)
2. You can implement your own custom RDP channel (would work with `MSTSC` / `windows RDP` too if done right) that implements that behaviour
